src/java/map/FloatMask.java

```diff
@@ -422,6 +422,116 @@ public FloatMask setSize(int size) {
+        return newMask;
+    }
+
+    public FloatMask copyOutsideRange(BinaryMask other) {
+        FloatMask newMask = copy().removeValuesInsideOf(other);
+        return newMask;
+    }
+
+    public FloatMask smoothWithinSpecifiedDistanceOfEdgesOf(BinaryMask other, int distance) {
+        for (int x = 0; x < distance; x = x + 2) {
+            replaceValuesInRangeWith(other.getAreasWithinSpecifiedDistanceOfEdges(x + 1), copy().smooth(1));
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask reduceValuesOnIntersectingSmoothingZones(BinaryMask avoidMakingZonesHere) {
+        FloatMask newMask = copy().smooth(34).subtract(copy()).subtract(avoidMakingZonesHere, 1f);
+        BinaryMask Zones = newMask.copy().removeValuesInRange(0f, 0.5f).smooth(2).convertToBinaryMask(0.5f, 1f).inflate(34);
+        BinaryMask newMaskBase = convertToBinaryMask(1f, 1f).deflate(3).minus(Zones.copy().invert());
+        newMask.init(newMaskBase, 0, 1).smooth(4).clampMax(0.35f).add(newMaskBase, 1f).smooth(2).clampMax(0.65f).add(newMaskBase, 1f).smooth(1).add(newMaskBase, 1f).clampMax(1f);
+        replaceValuesInRangeWith(Zones, newMask).smoothWithinSpecifiedDistanceOfEdgesOf(Zones, 30);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public BinaryMask convertToBinaryMask(float minValueToConvert, float maxValueToConvert) {
+        BinaryMask newMask = new BinaryMask(this.copy().removeValuesOutsideOfRange(minValueToConvert, maxValueToConvert), minValueToConvert, random.nextLong());
+        VisualDebugger.visualizeMask(this);
+        return newMask;
+    }
+
+    public FloatMask getDistanceFieldForRange(float minValue, float maxValue) {
+        convertToBinaryMask(minValue, maxValue).getDistanceField();
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask removeAreasOutsideOfSpecifiedIntensityAndSize(int minSize, int maxSize, float minIntensity, float maxIntensity) {
+        FloatMask tempMask2 = copy().init(this.copy().convertToBinaryMask(minIntensity, maxIntensity).removeAreasOutsideOfSpecifiedSize(minSize, maxSize).invert(), 0f, 1f);
+        this.subtract(tempMask2).clampMin(0f);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask removeAreasOfSpecifiedIntensityAndSize(int minSize, int maxSize, float minIntensity, float maxIntensity) {
+        subtract(this.copy().removeAreasOutsideOfSpecifiedIntensityAndSize(minSize, maxSize, minIntensity, maxIntensity));
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask removeAreasOfSpecifiedSizeWithLocalMaximums(int minSize, int maxSize, int levelOfPrecision) {
+        for (int x = 0; x < levelOfPrecision; x++) {
+            removeAreasOfSpecifiedIntensityAndSize(minSize, maxSize, (1f -(float) x / (float) levelOfPrecision), 1f);
+        }
```

Comment:
why the for loop?

Reply:
so you can specify level of precision
it cleans up texturing spreading (such as behind ridges) at basically equidistant arbitrary levels
more precision = more arbitrary levels = cleaner/fuller/more precise removal